geodiff/src/drivers/sqlitedriver.cpp

```diff
@@ -519,18 +512,51 @@ static void handleUpdated( const std::string &tableName, const TableSchema &tbl,
     ChangesetEntry e;
     e.op = ChangesetEntry::OpUpdate;
 
+    bool hasUpdates = false;
     size_t numColumns = tbl.columns.size();
     for ( size_t i = 0; i < numColumns; ++i )
     {
       Sqlite3Value v1( sqlite3_column_value( statement.get(), static_cast<int>( i + numColumns ) ) );
       Sqlite3Value v2( sqlite3_column_value( statement.get(), static_cast<int>( i ) ) );
       bool pkey = tbl.columns[i].isPrimaryKey;
       bool updated = !valuesEqual( v1.value(), v2.value() );
+      if ( updated )
+      {
+        // Let's do a secondary check for some column types to avoid false positives, for example
+        // multiple different string representations could be used for a single datetime value,
+        // see "Time Values" section in https://sqlite.org/lang_datefunc.html
+        if ( tbl.columns[i].type == TableColumnType::DATETIME )
+        {
+          Sqlite3Stmt stmtDatetime;
+          stmtDatetime.prepare( db, "SELECT datetime(?) != datetime(?)" );
+          sqlite3_bind_value( stmtDatetime.get(), 1, v1.value() );
+          sqlite3_bind_value( stmtDatetime.get(), 2, v2.value() );
+          if ( SQLITE_ROW == sqlite3_step( stmtDatetime.get() ) )
+          {
+            updated = sqlite3_column_int( stmtDatetime.get(), 0 );
+          }
+        }
+
+        if ( updated )
+        {
+          hasUpdates = true;
+        }
+      }
```

Comment:
Whatabout implementing `bool operator::==(const Sqlite3Value& v1, const Sqlite3Value& v2)` to improve readability?

Reply:
good idea - added in d594981